src/components/Welcome/index.tsx

```diff
@@ -70,14 +69,9 @@ const WelcomeActions: FunctionComponent<WelcomeActionsProps> = ({
 }
 
 const Welcome: FunctionComponent<StepComponentBaseProps> = ({ nextStep }) => {
-  const { steps } = useSdkOptions()
+  const [, { findStep }] = useSdkOptions()
   const { translate } = useContext(LocaleContext)
-
-  const { options } = steps.find(
-    (step) => step.type === 'welcome'
-  ) as StepConfigWelcome
-
-  const { title, descriptions, nextButton } = options || {}
+  const { title, descriptions, nextButton } = findStep('welcome')?.options || {}
```

Comment:
It seems a bit strange to me that step components should try to find their own options. These options are only ever used in this component, so I think it would make more sense to pass them as props to each component when we create the list of steps components. Interested to hear the reasoning behind this approach.

Reply:
In the current codebase, there's nothing wrong to pass `option` down to the step component from `StepsRouter`, it's convenient to develop the component without having to manually look-up the corresponding `option`.

But there's another side effect: it tightly couples the _routers_ and _step components_ together, requires the `option` prop to be passed down the tree, which we can't assure its integrity (if a component in the middle modifies or forget to pass it down, the end component will receive a wrong `option` prop). Furthermore, my intention was to incrementally minimise the passing props, clean-up/refactor specific ones as contexts/hooks for on-demand usage.

On the other hand, we're accepting more and more external contribution, who don't actually know how the props passing mechanism work, hence this decoupling could help them to focus on developing a specific component without having to pay too much attention to _properly_ handle the passed props.

Of course, there're no perfect solutions, but IMO at this stage of the SDK, we should consider modularity, flexibility & scalability over convenience.

Reply:
Thanks for the explanation. I agree, we should prioritise modularity, flexibility and scalability. 
Going forward we also want to allow more flexibility of integration by allowing flows that include multiple steps of the same type but with different options, so at some point we should start gathering for that possibility.